### PR TITLE
Feature/register login validation

### PR DESCRIPTION
**Pull request by Laracrofttx**

### CODE REVIEW

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 1]
```diff
@@ -6,68 +6,95 @@
...
+			userManager = _userManager;
+			signInManager = _signInManager;
+		}
+
+		public async Task<IdentityResult> RegisterAsync(RegisterRequestModel model)
+		{
+			var userExists = await userManager.FindByEmailAsync(model.Email);
+
+			if (userExists != null)
+				throw new ArgumentException("User with such email already exists.");
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs:29]: Please add error handling for these Exceptions.

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 2]
```diff
@@ -6,68 +6,95 @@
...
-
-            return new JwtSecurityTokenHandler().WriteToken(token);
-        }
-    }
+	public class AccountService : IAccountService
+	{
+		private readonly UserManager<ApplicationUser> userManager;
+		private readonly SignInManager<ApplicationUser> signInManager;
+
+
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: Please remove extra line

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 3]
```diff
@@ -6,68 +6,95 @@
...
+			if (!IsValidEmail(model.Email))
+				throw new ArgumentException("Invalid email address format.");
+
+			var user = new ApplicationUser { UserName = model.Email, Email = model.Email };
+			var result = await userManager.CreateAsync(user, model.Password);
+
+			if (result.Succeeded)
+			{
+				await this.signInManager.SignInAsync(user, false);
+
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: Please remove extra line

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs [hunk 4]
```diff
@@ -6,68 +6,95 @@
...
+		private bool IsValidEmail(string email)
+		{
+			if (string.IsNullOrWhiteSpace(email))
+				return false;
+
+			var emailPattern = @"^(?("")("".+?(?<!\\)""@)|(([0-9a-z]((\.(?!\.))|[-!#\$%&'\*\+/=\?\^`\{\}\|~\w])*)(?<=[0-9a-z])@))(?(\[)(\[(\d{1,3}\.){3}\d{1,3}\])|(([0-9a-z][-\w]*[0-9a-z]*\.)+[a-z0-9][\-a-z0-9]{0,22}[a-z0-9]))$";
+			return Regex.IsMatch(email, emailPattern);
+		}
+	}
 }
```
Comment (ZdravkoBorisov) [inline CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs:95]: Please add new line

**Review by ZdravkoBorisov — CHANGES_REQUESTED** [review]

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs (hunk 1 above)
Comment (Laracrofttx) [reply CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs:29]: Done

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs (hunk 2 above)
Comment (Laracrofttx) [reply CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: Done

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs (hunk 3 above)
Comment (Laracrofttx) [reply CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs]: Done

File: CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs (hunk 4 above)
Comment (Laracrofttx) [reply CryptoWorld.News/CryptoWorld.News.Core/Services/AccountService.cs:95]: Done

**Review by VelinArbov — APPROVED** [review]

**Review by ZdravkoBorisov — APPROVED** [review]

**Pull request merged**